[MWCC]: mwcceppc file path behavior broken

#75 (wibo 0.6.14+) causes issues with mwcceppc paths that worked in previous versions. This causes issues on all Linux environments. 

Our setup uses assetrip + assetinc to rip binary assets from the source binaries into build/include/assets/... and we have `-i build/include` passed into mwcceppc.exe. Since the issue started occurring with wibo 0.6.14, the issue must be these lines: https://github.com/decompals/wibo/pull/75/files#diff-35916b9fc9415b9ed5c6c9e5cb55c16f9c61754053de1ac138aa4456378610dfR685-R691

Example error:
```
[422/1595] CC build/iconv/src/static/version.c
FAILED: build/src/static/version.o 
/opt/devkitpro/devkitPPC/bin/powerpc-eabi-cpp -M build/iconv/src/static/version.c -MF build/src/static/version.o.d -nostdinc -I include -I tools/ppcdis/include -I build/include -I /N64_SDK/ultra/usr/include && wibo tools/1.3.2/mwcceppc.exe -O4 -char unsigned -fp hard -inline on -enum int -sdata 0 -sdata2 0 -d _LANGUAGE_C -d F3DEX_GBI_2 -d NDEBUG -d DEBUG=0 -d MUST_MATCH -nostdinc -proc gekko -maxerrors 1 -Cpp_exceptions off -I- -i include -i tools/ppcdis/include -i build/include -ir /N64_SDK/ultra/usr/include -c build/iconv/src/static/version.c -o build/src/static/version.o
### mwcceppc.exe Compiler:

User break, cancelled...
#    File: build\iconv\src\static\version.c
# -----------------------------------------
#       4: #include "assets/__Creator__.inc"
#   Error:                                 ^
#   the file 'assets/__Creator__.inc' cannot be opened
#   Too many errors printed, aborting program
[434/1595] Order floats config/rel.yml 806432c0 806432c8
ninja: build stopped: subcommand failed.
Dumping dump/forest_1st.arc
Dumping dump/forest_2nd.arc
Error running Ninja build
```

--- a/dll/kernel32.cpp
+++ b/dll/kernel32.cpp
@@ -701,16 +701,10 @@ namespace kernel32 {
 			return false;
 		}
 
-		// If pattern is empty, just iterate
-		if (handle->pattern.empty()) {
-			handle->it++;
-			return handle->it != std::filesystem::directory_iterator();
-		}
-
 		// Look for a matching file with the pattern
 		while (handle->it != std::filesystem::directory_iterator()) {
 			std::filesystem::path path = *handle->it;
-			if (fnmatch(handle->pattern.c_str(), path.filename().c_str(), 0) == 0) {
+			if (!handle->pattern.empty() && fnmatch(handle->pattern.c_str(), path.filename().c_str(), 0) == 0) {
 				return true;
 			}
 			handle->it++;

